Extend timer interface

mkykadir opened the pull request:
Include more (useful) interfaces from Vibe.d's timer

- `createTimer`: Creating timer without arming
- `pending`: State of the timer after arming (if waiting to be fired)

Commit:
Timer: Add `createTimer` interface
for enabling timer creation without arming

Commit:
Timer: Add `pending` interface
for checking if timer waiting to expire

mkykadir commented on source/geod24/LocalRest.d:
```diff
@@ -567,6 +587,12 @@ public final class Timer
+    /// True if timer is yet to fire
+    @property bool pending () @safe nothrow
+    {
+        return this._pending;
```
Maybe `&&` with `running` or setting `_pending` to `false` is required at L558

hewison-chris replied:
I would go with the second option so the state of `this.pending` is updated as it could be accessed not using this property and have the wrong state.

Geod24 commented:
I don't understand why we need `createTimer` ? The call site will have to select which function to call anyway, so it's not an abstraction.

mkykadir commented:
Just for convenience actually. One might want to create a timer in ctor and later manage it. `setTimer` creates and arms a timer which might be limiting for some cases. (Reentrant functions)

hewison-chris reviewed (changes requested):
LGTM, small addition as you suggested.

Commit:
Timer: Make interfaces and require handler to be `@safe` and `nothrow`
since there are no handling mechanism for exceptions
occurring inside the handler

hewison-chris reviewed (commented):
LGTM
